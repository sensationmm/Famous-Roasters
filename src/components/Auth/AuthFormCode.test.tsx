import { MockedProvider } from '@apollo/client/testing'
import { render, waitFor } from '@testing-library/react'
import React from 'react'

import { AuthFormCode } from '.'

describe('Auth code input custom auth component', () => {
  const snippet = () => (
    <form name="testAuthFormItemInput">
      <AuthFormCode screenKey="forgotPassword" onChange={() => null} />
    </form>
  )

  it('Renders correctly', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>{snippet()}</MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
