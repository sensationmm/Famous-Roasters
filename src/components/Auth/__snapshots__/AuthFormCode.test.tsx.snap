// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth code input custom auth component Renders correctly 1`] = `
<div>
  <form
    method="POST"
  >
    <label
      class="relative"
    >
      <span
        class="absolute -top-6 left-3 px-1 bg-white"
      >
        <span
          class="text-sm leading-5"
        >
          auth.forgotPassword.code.label
        </span>
      </span>
      <input
        class="w-full border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
        data-testid="code"
        placeholder="auth.forgotPassword.code.placeholder"
        type="text"
        value=""
      />
    </label>
  </form>
</div>
`;
